tests/e2e/vmgroups_test.go

```diff
@@ -0,0 +1,307 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This test is going to create volume on the fresh testbed very first time.
+// After installing vmdk volume plugin/driver, volume creation should not be
+// failed very first time.
+
+// This test is going to cover the issue reported at #656
+// TODO: as of now we are running the test against photon vm it should be run
+// against various/applicable linux distros.
+
+package e2e
+
+import (
+	"os"
+	"time"
+	. "gopkg.in/check.v1"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/dockercli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/ssh"
+	"github.com/vmware/docker-volume-vsphere/tests/constants/admincli"
+)
+
+const (
+	volName1                 = "_test_vol_1"
+	volName2                 = "_test_vol_2"
+	defaultVg                = "_DEFAULT"
+	defcname                 = "def_test_ctr"
+	defVgName                = "def_test_vg"
+)
+
+type VmGroupTest struct {
+	host string
+	vmgroup string
+	datastore1 string 
+	datastore2 string
+	cname string
+	dockerHosts []string
+}
+
+var _ = Suite(&VmGroupTest{})
+
+func curTime() string {
+	return time.Now().Format(time.UnixDate)
+}
+
+func (vg *VmGroupTest) SetUpSuite(c *C) {
+	vg.dockerHosts = []string{os.Getenv("VM1"), os.Getenv("VM2"), os.Getenv("VM3")}
+	vg.host = os.Getenv("ESX")
+	vg.vmgroup = os.Getenv("VMGROUP")
+	vg.datastore1 = os.Getenv("DS1")
```

Comment:
I would suggest letting code/util to retrieve available datastore (https://github.com/vmware/docker-volume-vsphere/pull/1286#discussion_r118540539) 

Rather than following stricter approach, we can let testcase author to decide which datastore to use from the available in the inventory.